web/source/osk/visualKeyboard.ts

```diff
@@ -172,53 +183,46 @@ namespace com.keyman.osk {
     }
 
     public postInsert(): void { }
-
+ 
+    /**
+     * The configured width for this VisualKeyboard.  May be `undefined` or `null`
+     * to allow automatic width scaling. 
+     */
     get width(): number {
-      if(this._width) {
-        return this._width;
-      } else {
-        this.loadSizeFromCookie();
-        return this.width;
-      }
+      return this._width;
     }
 
+    /**
+     * The configured height for this VisualKeyboard.  May be `undefined` or `null`
+     * to allow automatic height scaling. 
+     */
     get height(): number {
-      if(this._height) {
-        return this._height;
-      } else {
-        this.loadSizeFromCookie();
-        return this.height;
-      }
+      return this._height;
     }
 
-    protected loadSizeFromCookie() {
-      let keyman = com.keyman.singleton;
-      let util = keyman.util;
-
-      // If no prior cookie exists, it merely returns an empty object / cookie.
-      var c = util.loadCookie('KeymanWeb_OnScreenKeyboard');
-      var newWidth: number, newHeight: number;
-
-      // Restore OSK size - font size now fixed in relation to OSK height, unless overridden (in em) by keyboard
-      newWidth=util.toNumber(c['width'], 0.333 * screen.width); // Default - 1/3rd of screen's width.
+    get fontSize(): ParsedLengthStyle {
+      return this._fontSize;
+    }
 
-      if(newWidth < 0.2*screen.width) {
-        newWidth = 0.2*screen.width;
-      } else if(newWidth > 0.9*screen.width) {
-        newWidth=0.9*screen.width;
-      }
-  
-      // Default height decision made here: 
-      // https://github.com/keymanapp/keyman/pull/4279#discussion_r560453929
-      newHeight=util.toNumber(c['height'], 0.333 * newWidth);
+    set fontSize(value: ParsedLengthStyle) {
+      this._fontSize = value;
+      this.kbdDiv.style.fontSize = value.styleString;
+    }
 
-      if(newHeight < 0.15*screen.height) {
-        newHeight = 0.15 * screen.height;
-      } else if(newHeight > 0.5*screen.height) {
-        newHeight=0.5*screen.height;
-      }
+    /**
+     * Uses fixed scaling for widths of internal elements, rather than relative,
+     * percent-based scaling.
+     */
+    public get usesFixedWidthScaling(): boolean {
+      return this.device.touchable && !this.isStatic;
+    }
 
-      this.setSize(newWidth, newHeight);
+    /**
+     * Uses fixed scaling for heights of internal elements, rather than relative,
+     * percent-based scaling.
+     */
+    public get usesFixedHeightScaling(): boolean {
+      return this.device.touchable && !this.isStatic;
     }
```

Comment:
Up 'til now, desktop keyboards and documentation 'keyboards' have always used relative scaling, while touch keyboards have always used fixed (pixel-based) scaling.  Once the full OSK-Core redesign is implemented, these rules won't necessarily hold any longer.

Should someone want a documentation keyboard "rendered" at a fixed size, these will (once fully integrated) allow even a desktop keyboard to be rendered with precise pixel padding and width.  Alternatively, a live touch keyboard could be rendered with percent-based internal layout styling.